src/net_processing.cpp

```diff
@@ -1986,8 +1982,7 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
                 return true;
             pindex = (*mi).second;
 
-            if (!chainActive.Contains(pindex) &&
-                !StaleBlockRequestAllowed(pindex, chainparams.GetConsensus())) {
+            if (!StaleBlockRequestAllowed(pindex, chainparams.GetConsensus())) {
```

Comment:
So it was missing `IsValid(BLOCK_VALID_SCRIPTS)` here?

Reply:
"Missing" is a strong word given there isn't exactly a spec for the P2P protocol, but, yes, I'd say it should be there.

Reply:
Should be tagged backport then?

Reply:
Meh, dont think it matters too much.